### PR TITLE
Bump BPMN deps

### PR DESCRIPTION
**Pull request by nikku**
Closes https://github.com/camunda/camunda-modeler/issues/4035
Closes https://github.com/camunda/camunda-modeler/issues/3070
Closes https://github.com/camunda/camunda-modeler/issues/4070
Closes https://github.com/camunda/camunda-modeler/issues/4051
Closes https://github.com/camunda/camunda-modeler/issues/4037
Closes https://github.com/camunda/camunda-modeler/issues/3991
Closes https://github.com/camunda/camunda-modeler/issues/3895

### CODE REVIEW

**Comment by philippfromme** [comment]
> This PR is currently blocked as it fails locally (and in the distribution) with the following error message:
> 
> ```
> ERROR client omit is not a function
>     at ../node_modules/bpmnlint-plugin-camunda-compat/index.js (webpack://camundanode_modules/bpmnlint-plugin-camunda-compat/index.js:37:0)
>     at __webpack_require__ (webpack://camunda-modeler-client/webpack/bootstrap:19:0)
>     at __webpack_require__.t (webpack:/camunda-modeler-client/webpack/runtime/create fake namespace object:10:0)
>     at async (webpack://camundanode_modules/@camunda/linting/lib/Linter.js:145:0)
>     at async (webpack://camundanode_modules/@camunda/linting/lib/Linter.js:73:0)
>     at async (webpack://camunda-modeler-client/src/app/App.js:945:0)
>     at async file:///home/nikku/camunda/projects/bpmn.io/camunda-modeler/client/build/bundle.js:78366:14
> ```

Weird. We've been using `omit` in there for forever. 🤔

**Comment by nikku** [comment]
Found during integration testing the changes: https://github.com/camunda/camunda-modeler/issues/4098.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,19 +6,52 @@ All notable changes to the [Camunda Modeler](https://github.com/camunda/camunda-
...
+* `FEAT`: allow users to re-open recently used files ([#3917](https://github.com/camunda/camunda-modeler/issues/3917))
+* `FEAT`: confirm on unsaved files before reload ([#3886](https://github.com/camunda/camunda-modeler/issues/3886), [#3240](https://github.com/camunda/camunda-modeler/issues/3240))
+* `FEAT`: rework report feedback widget ([#4086](https://github.com/camunda/camunda-modeler/issues/4086))
+* `FEAT`: improve resizer and popup editor toggle interaction ([#3895](https://github.com/camunda/camunda-modeler/issues/3895))
+* `FIX`: correct user forum link ([#4085](https://github.com/camunda/camunda-modeler/issues/4085))
+* `CHORE`: track popup editor usage ([#4004](https://github.com/camunda/camunda-modeler/issues/4004))
 * `DEPS`: update to `@bpmn-io/form-js@1.7.0`
 * `DEPS`: update to `@camunda/form-linting@0.15.0`
 * `DEPS`: update to `@camunda/form-playground@0.13.0`
+* `DEPS`: update to `bpmn-js@16.4.0`
```
Comment (philippfromme) [inline CHANGELOG.md:22]: Some dependencies seem outdated. For example, bpmn-js latest is [v16.5.0](https://www.npmjs.com/package/bpmn-js?activeTab=versions). Are we okay with that?

Comment (nikku) [reply CHANGELOG.md:22]: Yes we are.

Comment (nikku) [reply CHANGELOG.md:22]: [That one](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#1650) arrived after the code freeze, and hence did not make it.

Comment (nikku) [reply CHANGELOG.md:22]: I'm not aware of other dependencies that are outdated.

Comment (philippfromme) [reply CHANGELOG.md:22]: 👍🏻

**Review by philippfromme — APPROVED** [review]
Good job! 👍🏻

**Pull request merged**